### PR TITLE
CBG-542 SG-specific CFG implementation

### PR DESCRIPTION
**Pull request by adamcfraser**
Adds cbgt CFG implementation based on an existing gocb bucket, with integration to caching feed.

Ready for review, but requires integration test verification.

### CODE REVIEW

File: base/dcp_sharded_cfg.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+func (c *CfgSG) Get(cfgKey string, cas uint64) (
+	[]byte, uint64, error) {
+
+	InfofCtx(c.loggingCtx, KeyDCP, "cfg_sg: Get, key: %s, cas: %d", cfgKey, cas)
+	bucketKey := sgCfgBucketKey(cfgKey)
+	var value []byte
+	casOut, err := c.bucket.Get(bucketKey, &value)
+	if err != nil && err != gocb.ErrKeyNotFound {
+		InfofCtx(c.loggingCtx, KeyDCP, "cfg_sg: Get, ErrKeyNotFound key: %s, cas: %d", cfgKey, cas)
```
Comment (bbrks) [inline base/dcp_sharded_cfg.go]: Not a KeyNotFound error

```suggestion
		InfofCtx(c.loggingCtx, KeyDCP, "cfg_sg: Get Error key: %s, cas: %d err:%s", cfgKey, cas, err)
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Mostly looks good. Left comments on a couple of things

File: base/dcp_sharded_cfg.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+// NewCfgCB returns a Cfg implementation that reads/writes its entries
+// from/to a couchbase bucket, using DCP streams to subscribe to
+// changes.
+//
+// urlStr: single URL or multiple URLs delimited by ';'
+// bucket: couchbase bucket name
+func NewCfgSG(bucket *gocb.Bucket) (*CfgSG, error) {
+
+	cfgContextID := fmt.Sprintf("%s-%s", MD(bucket.Name()), DCPImportFeedID)
```
[the remark this answers is not in the document]
Comment (adamcfraser) [reply base/dcp_sharded_cfg.go]: Makes sense - will standardize into a helper function (this is being done in a few places) and make those changes.

File: base/dcp_sharded_cfg.go (hunk 1 above)
Comment (adamcfraser) [reply base/dcp_sharded_cfg.go]: Thanks - will fix.

File: base/dcp_sharded_cfg.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamcfraser) [reply base/dcp_sharded_cfg.go]: Good catch

**Review by bbrks — APPROVED** [review]

**Pull request merged**